redux/react/ConnectMacro.hx

```diff
@@ -240,6 +237,17 @@ class ConnectMacro
 					case FFun(f):
 						if (f.args.length < 1) f.args.push(propsArg);
 						if (f.args.length < 2) f.args.push(contextArg);
+
+						var updateStateExpr = macro {
+							if (state == null)
```

Comment:
It looks like we're generating redundant `__state = mapState(context.store.getState(), props)` compiled in the resulting JS.